### PR TITLE
Getting started with neural network in PyTorch

### LINKED ISSUE #6905: Getting started with neural network in PyTorch
#### Proposed title of article
Getting started with neural network in PyTorch

#### Proposed article introduction
Neural networks are a set of algorithms, modeled loosely after the human brain, that are designed to recognize patterns. They interpret sensory data through a kind of machine perception, labeling or clustering raw input.

Without using a perceptron or any other hidden layers, we'll construct a single-layer neural network from scratch. It will have two layers: an input layer with training examples and synapses or weights, and neurons, and an output layer with the right responses.

#### Key takeaways 
- Installation of the required libraries
- Building a training model
- Adding learning iterations

#### Article quality
In this article, we will build a neural network that can only recognize very simple patterns on a very small data set. The article will be unique as I will expand it by trying to teach the AI to recognize something in pictures.

#### References
N/A

### PR DESCRIPTION
**Pull request by ivymugo**
## Our **Peer Reviewer and Content Moderator teams do NOT provide any revisions services.** All revisions and edits should be completed on your own forked repo (as to not take up room in the queue) in order for our team to review them all in a timely manner. 

**ANY ARTICLE SUBMITTED WITH GLARING ERRORS WILL BE IMMEDIATELY CLOSED.**

>As a rule of thumb - please be sure to only submit articles (pull requests) that are fully polished and ready to be published. Be sure to go through our resources documents for extra and 3rd party (vetted) resources to help improve overall technical writing.

> If you have not already, please go over our [Resources Page](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md) for more tips and tricks on how to improve your overall technical writing so reviews are swifter, increase the chance of a payout and provide more value to our readers.

> For first-time contributors and for more details on our submission guidelines, **see our [Contributing Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**.

#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is **600x400 pixels**, under **300KB in size**, and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?

This closes #6905

> Finally, delete the article checklist notes in blockquotes and submit your PR. We look forward to reviewing your article.

### CODE REVIEW

**Comment by louisefindlay23** [comment]
Plagiarism check complete

File: content/articles/getting-started-with-neural-network-in-pytorch/index.md [hunk 1]
```diff
@@ -0,0 +1,287 @@
+Neural networks are an algorithmic set of rules for discovering patterns in data. Using a neural network model will help us understand how our brains process information in the same way. The networks are constructed from discrete pieces that resemble neurons and are often called "neurons" or "units." There are a certain number of weighted inputs for each unit.
```
Comment (mercymeave) [inline content/articles/getting-started-with-neural-network-in-pytorch/index.md:1]: When you talk of the same, way what way ; mentioned before;  are you referring to?

File: content/articles/getting-started-with-neural-network-in-pytorch/index.md [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+### Table of Contents
+- [Introduction to Pytorch](#introduction-to-pytorch)
+- [Creating a neural network](#creating-a-neural-network)
+- [Training the neural network](#training-the-neural-network)
+- [Conclusion](#conclusion)
+
+Nodes in a neural network may be divided into three types, namely:
+- Input nodes - These nodes bring in data from the outside world. The Input nodes do nothing more than transmitting data to the hidden nodes; they do not undertake any computations.
+- Hidden nodes - These nodes are the ones that perform computations. They are the ones that are responsible for learning the weights of the connections between the input nodes and the output nodes. 
+- Output nodes - These nodes are the ones that are responsible for the final output. They are the ones that are responsible for the final decision.
```
Comment (mercymeave) [inline content/articles/getting-started-with-neural-network-in-pytorch/index.md:19]: Consider using simple sentences, something like "The input node transmit data from the external world into the hidden nodes."

File: content/articles/getting-started-with-neural-network-in-pytorch/index.md [hunk 3]
```diff
@@ -0,0 +1,287 @@
...
+
+### Introduction to Pytorch
+Pytorch is a [Python](https://pytorch.org/) library that allows us to create neural networks. It is a high-level library that enables us to build neural networks straightforwardly. It is also a prevalent library for deep learning.
```
Comment (mercymeave) [inline content/articles/getting-started-with-neural-network-in-pytorch/index.md:22]: There is repetition here. You already talked of pytorch  in the introduction.

File: content/articles/getting-started-with-neural-network-in-pytorch/index.md [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+
+#### Install Pytorch
+To install Pytorch, we need to install the [Python](https://www.python.org/) library. We can do this by using the command:
```
Comment (mercymeave) [inline content/articles/getting-started-with-neural-network-in-pytorch/index.md:25]: Here you say we install python but the command below it install, PyTorch, won't this confuse the reader?

File: content/articles/getting-started-with-neural-network-in-pytorch/index.md [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+```
+
+### Creating a neural network
+Pytorch has an `nn` module that contains all the building blocks for neural networks. The `nn` module contains the following submodules:
+- `nn.Module` - This is a base class for all the other modules.
+- `nn.Linear` - This is a fully connected layer and it takes in a matrix of inputs and outputs a matrix of outputs.
+- `nn.Sigmoid` - This is an activation function. It also takes in a matrix of inputs and outputs a matrix of outputs.
+- `nn.MSELoss` - This is a loss function. It takes in a matrix of outputs and targets and outputs a scalar.
+
+Let's begin by creating a neural network!
```
Comment (mercymeave) [inline content/articles/getting-started-with-neural-network-in-pytorch/index.md:38]: Don't mix up words. Explain every module clearly. Besides, use complete sentences 
.

File: content/articles/getting-started-with-neural-network-in-pytorch/index.md [hunk 6]
```diff
@@ -0,0 +1,287 @@
...
+import torchvision.datasets as datasets
+import torchvision.transforms as transforms
+```
+
+Let's go through them one by one to see what they do:
+
+```python
+import torch
+```
+This line imports the `torch` library. It is a high-level library that allows us to create neural networks. It is also a popular library for deep learning.
```
Comment (mercymeave) [inline content/articles/getting-started-with-neural-network-in-pytorch/index.md:58]: Repetition .

File: content/articles/getting-started-with-neural-network-in-pytorch/index.md [hunk 7]
```diff
@@ -0,0 +1,287 @@
...
+import torch.nn as nn
+```
+
+Essentially, this imports the `nn` module from the `torch` library. This module contains all the building blocks for neural networks.
+
+```python
+import torch.nn.functional as F
+```
+
+This module has the `F` function used to create the activation functions.
```
Comment (mercymeave) [inline content/articles/getting-started-with-neural-network-in-pytorch/index.md:70]: What is an activation function? How is it used in the neural networks? Explain these clearly and do it for every other module. Don't just do the brief explanation.

File: content/articles/getting-started-with-neural-network-in-pytorch/index.md [hunk 8]
```diff
@@ -0,0 +1,287 @@
...
+import torchvision.datasets as datasets
+```
+
+This module contains all the datasets we can use to train our neural networks.
+
+```python
+import torchvision.transforms as transforms
+```
+
+This module contains all the transformations we can use to preprocess our data.
```
Comment (mercymeave) [inline content/articles/getting-started-with-neural-network-in-pytorch/index.md:94]: What are `transformations` and why do we need them?

File: content/articles/getting-started-with-neural-network-in-pytorch/index.md [hunk 9]
```diff
@@ -0,0 +1,287 @@
...
+class Net(nn.Module):
+```
+
+In this line, we are creating a class called `Net`. This class is derived from the `nn.Module` class. This class is the base class for all the other modules.
+
+```python
+def __init__(self, input_size, num_classes):
+```
+The input size will be 784 since the amnesty image is 28x28 pixels. The `num_classes` will be ten since we want to classify the images into ten categories.
+
```
Comment (mercymeave) [inline content/articles/getting-started-with-neural-network-in-pytorch/index.md:125]: I am wondering where the image came from. Every step should be linked to another for maximum value.

**Review by mercymeave — CHANGES_REQUESTED** [review]
Hello, thanks for submitting your first article to Section.io.

However, your article has a lot of mistakes, grammatical errors, repetition and improper sentences.

I noticed I would take quite a long time doing the edits. I, therefore, recommend that you work on the changes to perfect the article.

**Comment by mercymeave** [comment]
I will give you more time to work on this. Create a new PR and ping me once you are done.
